### DISCUSSION TITLE
Is it normal for some games to have duplicated entries?

### DISCUSSION THREAD

**Discussion by mainrs** [mtkennerly/ludusavi#343] [Q&A] [answered]
I have two paths that I track changes in:

```
/home/me/.local/share/Steam
/home/me/.var/app/net.lutris.Lutris/data/lutris
```

When I click backup, there are multiple games that have overlapping files and I don't really understand why. Pressing on the `duplicates` tag filters the list and shows what games are affected. I have two groups, each containing two games with duplicated content.

Here are two screenshots of them:

![Screenshot_20240520_142610](https://github.com/mtkennerly/ludusavi/assets/5113257/c7750a9d-34fe-4878-b79d-8ac1d2738a97)

The second group has a long list of files:

![Screenshot_20240520_142652](https://github.com/mtkennerly/ludusavi/assets/5113257/58648a8a-1236-4410-99e4-ba3b7723a5cf)
![image](https://github.com/mtkennerly/ludusavi/assets/5113257/db7da1ab-2205-421d-9616-0abeab8be668)

It this normal for Windows games that contain registry paths? How does restore/backup work here then? I thought that it would only backup the keys that are required by the game. But the file sizes are the same; I'd assume it does a backup of the whole registry.

**Comment by mtkennerly** [comment]
Hi!

For the first case, Ludusavi does back up the whole `.reg` file for Wine prefixes, since it assumes that the prefix will only have one game. I do agree with you that it would be better to save just the relevant keys, though. Would you mind sharing a copy of those `system`, `user`, and `userdef` files from a game? I'd like to double check them for testing.

For the second case, here's some extra context:

* On PCGamingWiki, a game can have a primary Steam ID (`steam appid`) and several secondary IDs (`steam appid side`) for DLC and related content.
* In the article for `The Outer Worlds`, one of the secondary Steam IDs is 1920490.
* In the article for `The Outer Worlds: Spacer's Choice Edition`, the primary Steam ID is 1920490.
* When Ludusavi scans a game, it checks the `compatdata` folders for the primary and all secondary Steam IDs.

I think the easiest solution would be to ignore secondary IDs that are also the primary ID of another game.

**Reply by mainrs** [reply]
Thank you for the reply!

> I think the easiest solution would be to ignore secondary IDs that are also the primary ID of another game.

This sounds sensible to do. In my case, `The Outer Worlds` is/was the base game with separate DLCs. `Spacer's Choice Edition` is the GOTY edition. 

>  Would you mind sharing a copy of those system, user, and userdef files from a game? I'd like to double check them for testing.

I can do that! I have to double check first if they contain some private data (game keys etc.).

**Reply by mtkennerly** [reply, marked as answer]
Hey, just wanted to let you know that the primary/secondary ID issue has been fixed in the manifest already :)

For the Wine issue, I've opened a dedicated ticket to track it: https://github.com/mtkennerly/ludusavi/issues/354

**Reply by mainrs** [reply]
Ah, I forgot about the Wine registry files, sorry! I had to uninstall the game due to storage shortage :D but I'll re-install it over the weekend and send you the files starting next week! Creating a reminder right now :)
